modules/index/src/test/java/org/apache/ignite/internal/index/IndexAvailabilityControllerTest.java

```diff
@@ -0,0 +1,405 @@
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import org.apache.ignite.internal.catalog.CatalogManager;
+import org.apache.ignite.internal.catalog.CatalogTestUtils;
+import org.apache.ignite.internal.catalog.commands.AlterZoneParams;
+import org.apache.ignite.internal.catalog.commands.ColumnParams;
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogZoneDescriptor;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.lang.ByteArray;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.metastorage.Entry;
+import org.apache.ignite.internal.metastorage.impl.MetaStorageManagerImpl;
+import org.apache.ignite.internal.metastorage.impl.StandaloneMetaStorageManager;
+import org.apache.ignite.internal.replicator.ReplicaService;
+import org.apache.ignite.internal.storage.MvPartitionStorage;
+import org.apache.ignite.internal.storage.RowId;
+import org.apache.ignite.internal.storage.index.IndexStorage;
+import org.apache.ignite.internal.table.TableTestUtils;
+import org.apache.ignite.internal.table.distributed.index.IndexBuilder;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.internal.vault.inmemory.InMemoryVaultService;
+import org.apache.ignite.network.ClusterNode;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/** For {@link IndexAvailabilityController} testing. */
+public class IndexAvailabilityControllerTest extends BaseIgniteAbstractTest {
+    private static final String NODE_NAME = "test-node";
+
+    private static final String TABLE_NAME = "test-table";
+
+    private static final String COLUMN_NAME = "test-column";
+
+    private static final String INDEX_NAME = "test-index";
+
+    private final HybridClock clock = new HybridClockImpl();
+
+    private int partitions;
+
+    private VaultManager vaultManager;
+
+    private MetaStorageManagerImpl metaStorageManager;
+
+    private CatalogManager catalogManager;
+
+    private IndexBuilder indexBuilder;
+
+    private IndexAvailabilityController indexAvailabilityController;
+
+    @BeforeEach
+    void setUp() {
+        vaultManager = new VaultManager(new InMemoryVaultService());
```

Comment:
Let's make final all fields that allow this. For instance, `vaultManager`, `metaStorageManager`, `catalogManager`, `indexBuilder`, `indexAvailabilityController` can be made final and initialized via field initializers, if I'm not mistaken.

When a field is final, you don't need to bother for searching whether its value is redefined somewhere, so it becomes easier to reason about the code.

Reply:
Why not